lib/spack/spack/fetch_strategy.py

```diff
@@ -976,10 +981,13 @@ def _clone_src(self) -> None:
                 )
 
             with working_dir(dest):
-                checkout_args = ["checkout", self.commit]
+                checkout_args = ["checkout"]
+                checkout_kwargs = {}
                 if not debug:
-                    checkout_args.insert(1, "--quiet")
-                git(*checkout_args)
+                    checkout_args.extend(["--quiet", "--no-progress"])
+                    checkout_kwargs["error"] = str
+                checkout_args.append(self.commit)
+                git(*checkout_args, **checkout_kwargs)
```

Comment:
I think here you could do `error=str if ... else ...` instead of `**kwargs`, which helps static analysis tools.